### PR TITLE
Conduit subnet release on deletion

### PR DESCRIPTION
**Pull request by LionelJouin**
## Description

The conduit watcher in the IPAM is now deleting the conduit subnet when a conduit disappear from the conduit list sent by the NSP service.

## Issue link

Without this fix https://github.com/Nordix/Meridio/pull/349, new problems could appear when deleting and recreating a conduit.

## Checklist

- Purpose
    - [x] Bug fix
    - [ ] New functionality
    - [ ] Documentation
    - [ ] Refactoring
    - [ ] CI
- Test
    - [x] Unit test
    - [ ] E2E Test
    - [x] Tested manually
- Introduce a breaking change
    - [ ] Yes (description required)
    - [x] No
- Introduce changes in the Operator 
    - [ ] Yes (description required)
    - [x] No

### CODE REVIEW

File: pkg/ipam/trench/watcher.go [hunk 1]
```diff
@@ -21,48 +21,28 @@ import (
...
-		TrenchWatchers:             trenchWatchers,
-		ConfigurationManagerClient: configurationManagerClient,
-		logger:                     log.FromContextOrGlobal(ctx).WithValues("class", "ConduitWatcher"),
-	}
-	return cw, nil
-}
-
-func (cw *ConduitWatcher) Start() {
-	err := retry.Do(func() error {
+	_ = retry.Do(func() error {
```
Comment (zolug) [inline pkg/ipam/trench/watcher.go:39]: maybe some logs if the retry loop stops could be helpful (printing err as well)

Comment (LionelJouin) [reply pkg/ipam/trench/watcher.go:40]: I just added new logs in the retry loop.
The retry loop won't return any error,  `WithErrorIngnored` is used.

File: pkg/ipam/trench/watcher.go [hunk 2]
```diff
@@ -72,22 +54,46 @@ func (cw *ConduitWatcher) Start() {
...
+// on the differences between currentConduits and newConduits
+func SetConduits(ctx context.Context, tw TrenchWatcher, currentConduits map[string]struct{}, newConduits []*nspAPI.Conduit) {
+	if tw == nil {
+		return
+	}
+	logger := log.FromContextOrGlobal(ctx).WithValues("class", "ConduitWatcher")
+	// To add/update
+	for _, conduit := range newConduits {
+		// To add
+		_, err := tw.AddConduit(ctx, conduit.GetName())
```
[the remark this answers is not in the document]
Comment (LionelJouin) [reply pkg/ipam/trench/watcher.go:84]: right, maybe that is not the behavior we want to have.

**Comment by zolug** [comment]
Looks good.

**Review by zolug — APPROVED** [review]

**Pull request merged**